instrumentation/spring/spring-webmvc-5.3/library/src/main/java/io/opentelemetry/instrumentation/spring/webmvc/v5_3/HttpRouteSupport.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.webmvc.v5_3;
+
+import static java.util.Objects.requireNonNull;
+import static org.springframework.web.util.ServletRequestPathUtils.PATH_ATTRIBUTE;
+
+import io.opentelemetry.context.Context;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.Nullable;
+import javax.servlet.FilterConfig;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.web.context.ConfigurableWebApplicationContext;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.context.support.WebApplicationContextUtils;
+import org.springframework.web.servlet.DispatcherServlet;
+import org.springframework.web.servlet.HandlerExecutionChain;
+import org.springframework.web.servlet.HandlerMapping;
+import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping;
+import org.springframework.web.util.ServletRequestPathUtils;
+
+final class HttpRouteSupport {
+
+  private final AtomicBoolean contextRefreshTriggerred = new AtomicBoolean();
+  @Nullable private WeakReference<DispatcherServlet> dispatcherServlet;
+  @Nullable private volatile List<HandlerMapping> handlerMappings;
+  private volatile boolean parseRequestPath;
+
+  void onFilterInit(FilterConfig filterConfig) {
+    WebApplicationContext context =
+        WebApplicationContextUtils.getWebApplicationContext(filterConfig.getServletContext());
+    if (!(context instanceof ConfigurableWebApplicationContext)) {
+      return;
+    }
+
+    DispatcherServlet servlet = context.getBeanProvider(DispatcherServlet.class).getIfAvailable();
+    if (servlet != null) {
+      dispatcherServlet = new WeakReference<>(servlet);
+
+      ((ConfigurableWebApplicationContext) context)
+          .addApplicationListener(new WebContextRefreshListener());
+    }
+  }
+
+  // we can't retrieve the handler mappings from the DispatcherServlet in the onRefresh listener,
+  // because it loads them just after the application context refreshed event is processed
+  // to work around this, we're setting a boolean flag that'll cause this filter to load the
+  // mappings the next time it attempts to set the http.route
+  final class WebContextRefreshListener implements ApplicationListener<ContextRefreshedEvent> {
```

Comment:
just curious if this gets hit in the integration tests (I'm ok if not)

Reply:
It does; it has to be hit if the `http.route` is to be present in generated spans.